apps/web/src/app/core/router.service.ts

```diff
@@ -67,14 +86,14 @@ export class RouterService {
    * @param url URL being saved to the Global State
    */
   async persistLoginRedirectUrl(url: string): Promise<void> {
-    await this.stateService.setDeepLinkRedirectUrl(url);
+    await this.deepLinkRedirectUrlState.update((_) => url);
```

Comment:
🎨  suggestion: I think you can just skip the parameter all together
```suggestion
    await this.deepLinkRedirectUrlState.update(() => url);
```